src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2970,7 +2970,7 @@ private Variables GetOrCreateNestedFunctionVariables(Variables container, Method
                 }
 
                 VisitAlways(lambdaOrFunction.Body);
-                EnforceDoesNotReturn(syntaxOpt: null);
+                //EnforceDoesNotReturn(syntaxOpt: null);
                 EnforceParameterNotNullOnExit(null, this.State);
```

Comment:
This one too might be redundant, or lacks a test coverage.

Reply:
@RikkiGibson @jjonescz At least at the time I opened this PR, all tests passed with this line commented out. Not sure if it should just be left as-is and close this draft, or if it's worth investigation?